tools/clang/lib/SPIRV/InitListHandler.cpp

```diff
@@ -207,9 +198,13 @@ SpirvInstruction *InitListHandler::createInitForType(QualType type,
   if (type->isBuiltinType())
     return createInitForBuiltinType(type, srcLoc);
 
-  if (hlsl::IsHLSLVecType(type))
-    return createInitForVectorType(hlsl::GetHLSLVecElementType(type),
-                                   hlsl::GetHLSLVecSize(type), srcLoc);
+  QualType elemType;
+  uint32_t elemCount = 0;
+  if (isVectorType(type, &elemType, &elemCount))
+    return createInitForVectorType(elemType, elemCount, srcLoc);
+
+  if (isScalarType(type, &elemType))
```

Comment:
Add a comment `// the purpose of this check is for vectors of size 1 (for which isVectorType is false).`